### PR TITLE
Axis options

### PR DESCRIPTION
**Pull request by GraceGSy**
Changes in this PR:

**Axis options**
- Styling options added for axis main line, labels, title and ticks
- Positioning options for axis using hjust/vjust or x, w/y, h (in screen pixels) or x1, x2/y1, y2 (in screen pixels)
- An extra tick can be added at the origin of each axis for non-categorical variables, label for this extra tick can be toggled
- Minor adjustment to defaultFormat.js so that all output is in type `String`

**Nested axes**
- Axes can now be nested in subsections, see Multiline in sandbox

**Axis docs**
- Updated axis docs

**Feedback needed**
- ~~In order to fix the height/width of the x/y axes to 100px, a recursive function `getParents()` is used to implement an inverse this.$$transform~~
- ~~This may not be the best/most efficient method of converting screen pixels to local coordinates~~
- ~~Also: the extra tick comes with a label, that may sometimes be oddly formatted (see scatterplot). This implies that the user will have to manually set the formatting if the label turns out weird. Should we just remove the label for the extra tick at origin?~~

### CODE REVIEW

**Comment by luucvanderzee** [comment]
Nice work, lots of options to customize the axes!

Few things:

- Just to be clear, is there any reason that you want to use the `100px` value in particular? Or do you just want to be able to set any pixel value? In the first case, why 100? In the second case: the `getParents` function seems like an ok solution, but I don't really see where/how it currently works... which prop do you use to enter the width/height in pixels? Anyway, the one problem I think with this `getParents` function is that you are kind of doing double work. You are using it to take pixel coordinates by going all the way through the `CoordinateTree` just to get the local coordinates that these pixel coordinates correspond to. Which are then transformed back to pixel coordinates again with `$$transform`... I think that maybe we could have a more general solution for this problem (not just within the axes). Right now, `w` and `h` props are never allowed to take Strings, only numbers. We could add the functionality to allow pixel values at this level by allowing Strings of the format `'<Number>px'` (like `'100px'` to be passed to `w` and `h`. Then instead of using a recursive function to calculate the local coordinates to then convert those back to pixels again, we could just have some more sound logic in the `Rectangular` mixin to handle this (if `w` is a pixel value: first convert `x` from local to pixel value, then just calculate `x1` and `x2` in pixel values using the pixel value passed to `w`).  

- I saw that you converted all the `vgg-point`s back again into `vgg-symbol`s. I think maybe we didn't discuss this clearly last time, but I think we should just keep using `vgg-point` whenever we are dealing with points. If it is your goal to create a scatterplot where different symbols representing different categorical values, it makes sense to use `vgg-symbol`. But if I am making a scatterplot with points, I just want to use points, and not 'symbols of the type point'. So could you change those back? We could still refactor the `vgg-point` mark to actually just be a `vgg-symbol` internally, but then a `vgg-symbol` that is always a point and uses a `radius` instead of `size`, if you are worried about repeating logic (which was the original argument provided for transformating all points to symbols). I already made an issue about this (#50). 

- Maybe the label for the first tick could be toggled with another prop?

**Comment by GraceGSy** [comment]
Thanks for the feedback! Just to clarify the points raised:

- I'm using the `getParents()` function to compute `this._parentNodes` and then using that for the `getLocalX` and `getLocalY` functions. They're used in individual axis components, such as tickMin, tickMax, or converting this.x1 into local coordinates. In general, the width/height dimensions are entered using the inherited `Rectangular` props (so x1, x2, y1, y2 etc...). But I definitely agree that `getParents()` is doing extra work, so I don't really like it as a solution. Adding String processing logic to `Rectangular` sounds much better.

[Also to be honest, there is no real reason the axis needs to be any particular width/height since tick sizes and stroke widths are all customisable. I'm just arbitrarily using `100px` to set the size of the section enclosing the axis. Maybe this isn't the best implementation but I wanted to make it easier to calculate local coordinates for tick sizes in some cases.]

- I think I must have missed some `vgg-points` when merging. Which ones are still showing up as `vgg-symbol`? B/c I do agree that using radius with points is more intuitive, so if anything is still using `vgg-symbol` that would be unintentional...

- Yeah, an extra prop for first tick label sounds good

File: stories/sandbox/BarChart.vue [hunk 1]
```diff
@@ -55,35 +55,31 @@
             color="white"
           />
 
-          <vgg-point
+          <vgg-symbol
```
Comment (luucvanderzee) [inline stories/sandbox/BarChart.vue]: point -> symbol

File: stories/sandbox/NestedCoordinateSystem.vue [hunk 2]
```diff
@@ -27,7 +27,7 @@
           :scale-y="domains.y"
         >
 
-          <vgg-point
+          <vgg-symbol
```
Comment (luucvanderzee) [inline stories/sandbox/NestedCoordinateSystem.vue]: here again

**Comment by luucvanderzee** [comment]
Wait, if the axis is just inheriting from the section that it is in, why does the section need to be 100px? I don't really understand this point

File: docs/axes/cartesian.md [hunk 3]
```diff
@@ -2,27 +2,116 @@
...
+There are three options for positioning the x axis on the graph. The default position of the x-axis is at `vjust = 'b'` (bottom of parent section).
 
-### Custom Ticks
\ No newline at end of file
+When using `vjust`, the x axis defaults to a height of 100px in screen coordinates.
+
+| Prop | Required | Regular types    | Default   | Description                             | Unit(s)                |
+| ---- | -------- | ---------------- | --------- | --------------------------------------- | ---------------------- |
+| vjust| false    | [Number, String] | 'b'       | position of x axis                      | Number between 0 and 1 |
+| y    | false    | [Number]         | undefined | position of x axis                      | Screen pixels          |
```
Comment (luucvanderzee) [inline docs/axes/cartesian.md]: Shouldn't all these be in 'local coordinates' rather than 'screen pixels'?

File: src/components/Core/Section.vue [hunk 4]
```diff
@@ -194,8 +194,9 @@ export default {
   provide () {
     let $$transform = this.$$coordinateTree.getTotalTransformation(this.coordinateTreeBranchID)
     let $$coordinateTreeParent = this.coordinateTreeBranchID
-
-    return { $$transform, $$coordinateTreeParent }
+    let $$getLocalX = this.$$coordinateTree.getLocalX(this.coordinateTreeBranchID)
```
Comment (luucvanderzee) [inline src/components/Core/Section.vue]: Maybe these should be combined into one fuction, similar to `$$transform`?
Also does this work with polar coordinates?

File: src/components/Guides/YAxis.vue [hunk 5]
```diff
@@ -79,6 +108,103 @@
...
-  mixins: [BaseAxis]
+  mixins: [BaseAxis],
+
+  props: {
+    titleHjust: {
+      default: 'center'
+    },
+
+    titleVjust: {
+      default: 1.08
```
Comment (luucvanderzee) [inline src/components/Guides/YAxis.vue]: Why 1.08?

File: src/mixins/Guides/BaseAxis.js [hunk 6]
```diff
@@ -56,23 +229,65 @@ export default {
...
+    _scalingOptions () {
+      return this._parsedScalingOptions[2]
+    },
+
+    _parentNodes () {
+      return this.getParents(this.parentBranch, [this.parentBranch])
+    },
+
+    validX () {
+      return (this.x1 != undefined && this.x2 != undefined) || (this.x != undefined && this.w != undefined)
```
Comment (luucvanderzee) [inline src/mixins/Guides/BaseAxis.js]: Make sure that you always use `!==` instead of `!=`

File: src/components/Guides/YAxis.vue (hunk 5 above)
Comment (GraceGSy) [reply src/components/Guides/YAxis.vue]: Because it looks good XD

File: docs/axes/cartesian.md (hunk 3 above)
Comment (GraceGSy) [reply docs/axes/cartesian.md]: Now they are!

**Pull request merged**